uro98/assignment1/KToLast.java

```diff
@@ -0,0 +1,58 @@
+public class KToLast {
+
+	public static Object kToLastObject(SinglyLinkedList list, int k) {
+
+		SinglyLinkedList.Node node = list.first;
+		int length = 0;
+
+		// Find list length
+		while (node != null) {
+			length++;
+			node = node.next;
+		}
+
+		int target = length - k - 1;
+		int index = 0;
+		node = list.first;
+
+		// Get target node
+		while (index != target) {
+			node = node.next;
+			index++;
+		}
+
+		return node.data;
+	}
+
+	public static void main(String[] args) {
+		SinglyLinkedList list = new SinglyLinkedList(1);
+
+		SinglyLinkedList.Node second = list.new Node(2);
+		SinglyLinkedList.Node third = list.new Node(3);
+
+		list.first.next = second;
+		second.next = third;
+
+		System.out.println(kToLastObject(list, 1));
+	}
+}
+
+class SinglyLinkedList {
+
+	Node first;
+
+	public SinglyLinkedList(Object obj) {
+		first = new Node(obj);
+	}
+
+	class Node {
+
+		Object data;
```

Comment:
Just storing `Object` is not very convenient for users of the class as they have to remember what data they put inside and then cast it back to their specific type. Could you change code to use generics?